### DISCUSSION TITLE
Editors and IDEs support?

### DISCUSSION THREAD

**Discussion by 1e9t8m29** [titzer/virgil#102] [Ideas]
It seems there is none at the moment. I suggest adding at least syntax highlighting to CudaText: https://github.com/Alexey-T/CudaText

CudaText is inspired by SublimeText but is more lightweight and available for more platforms.

@titzer

**Comment by titzer** [comment]
There are modes for both `emacs` and `vim`. It would be great to have syntax highlighting support for more editors like Sublime and VSCode. (I had explored Sublime but my experiment has long since bit-rotted).

**Comment by 1e9t8m29** [comment]
> There are modes for both `emacs` and `vim`.

Where are they? I didn't find them on this repo.

> It would be great to have syntax highlighting support for more editors like Sublime and VSCode. (I had explored Sublime but my experiment has long since bit-rotted).

The CudaText developer is very friendly. You could go here: https://github.com/Alexey-T/CudaText/issues and create a ticket asking him to support Virgil.

**Comment by Alexey-T** [comment]
CudaText lexer exists already in addons-manager.

**Comment by titzer** [comment]
The emacs mode is in `bin/virgil-mode.el` and the vim mode is in `bin/dev/virgil-vim`.

**Comment by kalwalt** [comment]
@titzer i installed the vscode extension from `bin/dev/virgil-vscode` with the manual installation of the `.vsix` file. But i haven't found any documentation about this. Apart this, virgil is a great discovery!

**Reply by titzer** [reply]
There should probably be a page in `doc/` about this.
